scripts/install.sh

```diff
@@ -0,0 +1,233 @@
+else
+    if [ -e /etc/lsb-release ]; then
+        . /etc/lsb-release
+    fi
+fi
+
+if [ -z "${DISTRIB_ID}" ]; then
+    log INFO "Trying to identify using OSTYPE var $OSTYPE "
+    if [[ "$OSTYPE" == "linux"* ]]; then
+        DISTRIB_ID="$OSTYPE"
+        elif [[ "$OSTYPE" == "darwin"* ]]; then
+        DISTRIB_ID="$OSTYPE"
+        elif [[ "$OSTYPE" == "cygwin" ]]; then
+        DISTRIB_ID="$OSTYPE"
+        elif [[ "$OSTYPE" == "msys" ]]; then
+        DISTRIB_ID="$OSTYPE"
+        elif [[ "$OSTYPE" == "win32" ]]; then
+        DISTRIB_ID="$OSTYPE"
+        elif [[ "$OSTYPE" == "freebsd"* ]]; then
+        DISTRIB_ID="$OSTYPE"
+    else
+        log ERROR "Unknown DISTRIB_ID or DISTRIB_RELEASE."
+        exit 1
+    fi
+fi
+
+if [ -z "${DISTRIB_ID}" ]; then
+    log ERROR "Unknown DISTRIB_ID or DISTRIB_RELEASE."
+    exit 1
+fi
+
+log INFO "Distro Information as $DISTRIB_ID"
+
+# set distribution specific vars
+PACKAGER=
+SYSTEMD_PATH=/lib/systemd/system
+if [ "$DISTRIB_ID" == "ubuntu" ]; then
+    PACKAGER=apt
+    elif [ "$DISTRIB_ID" == "debian" ]; then
+    PACKAGER=apt
+    elif [[ $DISTRIB_ID == centos* ]] || [ "$DISTRIB_ID" == "rhel" ]; then
+    PACKAGER=yum
+    elif [[ "$DISTRIB_ID" == "darwin"* ]]; then
+    PACKAGER=brew
+    elif [[ "$DISTRIB_ID" == "win32" ]] || [[ "$DISTRIB_ID" == "msys" ]]; then
+    PACKAGER=choco
+else
+    PACKAGER=zypper
+    SYSTEMD_PATH=/usr/lib/systemd/system
+fi
+if [ "$PACKAGER" == "apt" ]; then
+    export DEBIAN_FRONTEND=noninteractive
+fi
+
+# Check JQ Processor and download if not present
+check_jq_processor_present() {
+    check_if_exists jq
+    if [[ $result != 0 ]]; then
+        install_tool jq
+    fi
+    jqversion=$(jq --version)
+    log INFO "Locally installed JQ Processor version is $jqversion"
+}
+
+check_kubectl_present() {
+    check_if_exists kubectl
+    if [[ $result != 0 ]]; then
+        install_tool kubectl
+    fi
+    kubectlversion=$(kubectl version --client=true -o json | jq ".clientVersion.gitVersion")
+    log INFO "Locally installed kubectl version is $kubectlversion"
+}
+
+check_dotnet_runtime_present() {
+    check_if_exists dotnet
+    # if dotnet doesn't exist install it
+    if [[ $result != 0 ]]; then
+        install_tool dotnet
+        return;
+    fi
+    #if dotnet exists, check the version required for b2k and install it.
+    dotnetruntimes=$(dotnet --list-runtimes)
+    if [[ -z "${dotnetruntimes}" || ! "${dotnetruntimes}" =~ '3.1'* ]]; then
+        install_tool dotnet
+    else 
+        log INFO "dotnet version is $(dotnet --version)"
+    fi
+}
+
+install_tool() {
+    log INFO "installing $1.."
+
+    case $1 in 
+
+        kubectl)
+            install_with_sudo kubectl
+            ;;
+        dotnet)
+            if [[ $OSTYPE == "darwin"* ]]; then
+                $PACKAGER tap isen-ng/dotnet-sdk-versions
+                install_with_sudo dotnet-sdk3-1-400 --cask
+            else 
+                install_with_sudo aspnetcore-runtime-3.1
```

Comment:
it will be dependency on that PR, we can update them once it is tested with dotnet 6, and everything looks good. My opinion that would take some time and we can't release this feature. @sabbour